app/src/main/java/com/github/libretube/ui/fragments/ChannelContentFragment.kt

```diff
@@ -0,0 +1,182 @@
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        _binding = FragmentChannelContentBinding.inflate(inflater, container, false)
+        return _binding!!.root
+    }
+
+    override fun setLayoutManagers(gridItems: Int) {
+        binding.channelRecView.layoutManager = GridLayoutManager(
+            requireContext(),
+            gridItems.ceilHalf()
+        )
+    }
+
+    private suspend fun fetchChannelNextPage(nextPage: String): String? {
+        val response = withContext(Dispatchers.IO) {
+            RetrofitInstance.api.getChannelNextPage(channelId!!, nextPage).apply {
+                relatedStreams = relatedStreams.deArrow()
+            }
+        }
+        channelAdapter?.insertItems(response.relatedStreams)
+        return response.nextpage
+    }
+
+    private suspend fun fetchTabNextPage(nextPage: String, tab: ChannelTab): String? {
+        val newContent = withContext(Dispatchers.IO) {
+            RetrofitInstance.api.getChannelTab(tab.data, nextPage)
+        }.apply {
+            content = content.deArrow()
+        }
+
+        searchChannelAdapter?.let {
+            it.submitList(it.currentList + newContent.content)
+        }
+        isLoading = false
+        return newContent.nextpage
+    }
+
+    override fun onConfigurationChanged(newConfig: Configuration) {
+        super.onConfigurationChanged(newConfig)
+        // manually restore the recyclerview state due to https://github.com/material-components/material-components-android/issues/3473
+        binding.channelRecView.layoutManager?.onRestoreInstanceState(recyclerViewState)
+    }
+
+    private fun loadChannelTab(tab: ChannelTab) = lifecycleScope.launch {
+        val response = try {
+            withContext(Dispatchers.IO) {
+                RetrofitInstance.api.getChannelTab(tab.data)
+            }.apply {
+                content = content.deArrow()
+            }
+        } catch (e: Exception) {
+            binding.progressBar.visibility = View.GONE
+            return@launch
+        }
+        withContext(Dispatchers.Main) {
+            nextPage = response.nextpage
+        }
+        val binding = _binding ?: return@launch
+        searchChannelAdapter = SearchChannelAdapter()
+        binding.channelRecView.adapter = searchChannelAdapter
+
+        searchChannelAdapter?.submitList(response.content)
+        binding.progressBar.visibility = View.GONE
+        binding.channelRecView.visibility = View.VISIBLE
+        isLoading = false
+    }
+
+    private fun loadNextPage(isVideo: Boolean, tab: ChannelTab) = lifecycleScope.launch {
+        try {
+            isLoading = true
+            if (isVideo) {
+                fetchChannelNextPage(nextPage ?: return@launch).let {
+                    nextPage = it
+                }
+            } else {
+                fetchTabNextPage(nextPage ?: return@launch, tab).let {
+                    nextPage = it
+                }
```

Comment:
```suggestion
                nextPage = fetchChannelNextPage(nextPage ?: return@launch)
            } else {
                nextPage = fetchTabNextPage(nextPage ?: return@launch, tab)
```

Reply:
Thank you @Bnyro for all the corrections, I will keep these in mind next time. Got to learn a lot from your corrections. I'll correct it and raise PR again.